corehq/messaging/whatsapputil.py

```diff
@@ -45,7 +45,7 @@ def get_template_hsm_parts(message_text):
     """The magic string users enter looks like: cc_wa_template:template_name:lang_code:{var1}{var2}{var3}
     """
     HsmParts = namedtuple("hsm_parts", "template_name lang_code params")
-    parts = message_text.split(":")
+    parts = message_text.split(":", maxsplit=3)
```

Comment:
Nit: could do something like this to clarify expectations a bit
```suggestion
    _, template_name, lang_code, params = message_text.split(":", maxsplit=3)
```

Reply:
yeah, that would have been a nicer way to do it (although `params` is optional so it would still need a try/except block).

Reply:
I'm a little late to the party. And this is more of a fun Python3 _whaddayaknow_ than an actionable suggestion:
```python
    _, template_name, lang_code, *params = message_text.split(":", maxsplit=3)
    if params:
        params = re.findall("{(.+?)}+", params[0])
```

This works because
```python
>>> message_text = 'cc_wa_template:template_name:lang_code:{var1}{var2}{var3}'
>>> _, template_name, lang_code, *params = message_text.split(":", maxsplit=3)
>>> params
['{var1}{var2}{var3}']
>>> message_text = 'cc_wa_template:template_name:lang_code'
>>> _, template_name, lang_code, *params = message_text.split(":", maxsplit=3)
>>> params
[]
```

Reply:
Cool! Thanks @kaapstorm